File: current_session/1509.py
"""
1509
"""
class Solution:
    def minDifference(self, nums: List[int]) -> int:
        if len(nums) <= 4: return 0
        nums.sort()
        l, r = 0, len(nums)-3
        res = float('inf')
        while r <= len(nums):
            res = min(res, nums[r-1]-nums[l])
            l, r = l+1, r+1
        
        return res

File: current_session/c++/1509.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  int minDifference(vector<int>& nums) {
    if (nums.size() <= 4) return 0;
    sort(nums.begin(), nums.end());
    int minDiff = INT_MAX;
    int l=0, r=nums.size()-4;
    while (r<nums.size()) {
      // cout << "l:" << l << "  " << nums[l] << endl;
      // cout << "r:" << r << "  " << nums[r] << endl;
      minDiff = min(minDiff, nums[r]-nums[l]);
      r++;
      l++;
    }
    return minDiff;
  }
};